src/components/inputs/text/ValidationInput/ValidationInput.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState } from 'react';
+
+import { ValidationObject } from '../../../../types';
+import ValidationList from '../../../cards/ValidationList/ValidationList';
+
+function ValidationInput({
+  wrapperClassName,
+  wrapperCustomStyle,
+  showValidationChecklist,
+  inputClassName,
+  inputCustomStyle,
+  placeholder,
+  disabled,
+  maxLength,
+  value,
+  setValue,
+  setIsValid,
+  validationPredicate,
+}: {
+  wrapperClassName?: string;
+  wrapperCustomStyle?: any;
+  showValidationChecklist?: boolean;
+  placeholder?: string;
+  maxLength?: number;
+
+  inputClassName?: string;
+  inputCustomStyle?: any;
+  disabled?: boolean; // disables input
+  value: string;
+  setValue: (text: string) => void;
+  setIsValid: (validity: boolean) => void;
+  validationPredicate: ((id: string) => Promise<ValidationObject>)[];
+}) {
+  const [validationResults, setValdidationResults] =
+    useState<ValidationObject[]>();
+
+  async function validationExecutor(id: string) {
+    setValue(id);
+    const validationResult: Promise<ValidationObject>[] = [];
+    validationPredicate.forEach((predicate) => {
+      validationResult.push(predicate(id));
+    });
+    const results = await Promise.all(validationResult);
```

Comment:
we'll want to use `Promise.allSettled` so it doesn't reject on the first one https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled